packages/core-flows/src/price-list/workflows/update-price-lists.ts

```diff
@@ -1,13 +1,38 @@
-import { PriceListDTO, UpdatePriceListWorkflowInputDTO } from "@medusajs/types"
-import { WorkflowData, createWorkflow } from "@medusajs/workflows-sdk"
-import { updatePriceListsStep } from "../steps"
+import { UpdatePriceListWorkflowInputDTO } from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { updatePriceListsStep, upsertPriceListPricesStep } from "../steps"
 
 type WorkflowInput = { price_lists_data: UpdatePriceListWorkflowInputDTO[] }
 
 export const updatePriceListsWorkflowId = "update-price-lists"
 export const updatePriceListsWorkflow = createWorkflow(
   updatePriceListsWorkflowId,
-  (input: WorkflowData<WorkflowInput>): WorkflowData<PriceListDTO[]> => {
-    return updatePriceListsStep(input.price_lists_data)
+  (input: WorkflowData<WorkflowInput>): WorkflowData<void> => {
+    upsertPriceListPricesStep(
+      transform({ input }, (data) => {
+        return data.input.price_lists_data.map((priceListsData) => {
+          const { prices = [], ...rest } = priceListsData
+
+          return {
+            price_list_id: rest.id,
+            prices,
+          }
+        })
+      })
+    )
+
+    updatePriceListsStep(
+      transform({ input }, (data) => {
+        return data.input.price_lists_data.map((priceListsData) => {
+          const { prices, ...rest } = priceListsData
+
+          return rest
+        })
+      })
+    )
```

Comment:
comment: I don't link this way of using the transforms, makes it hard to read. Please separate in "variable assignments"

Reply:
Pretty sure you don't need them here actually. You are passing workflow inputs. Should not need a transformer.